src/binary_agreement/binary_agreement.rs

```diff
@@ -121,7 +121,12 @@ impl<N: NodeIdT> BinaryAgreement<N> {
     }
 
     /// Sets the input value for Binary Agreement.
-    pub fn handle_input(&mut self, input: bool) -> Result<Step<N>> {
+    ///
+    /// If more than two thirds of validators input the same value, that will eventually be output.
+    /// Otherwise either output is possible.
+    ///
+    /// Note that if `can_input` returns `false`, it is already too late to affect the outcome.
+    pub fn set_input(&mut self, input: bool) -> Result<Step<N>> {
```

Comment:
Should this not be something like `propose`, `send_input`, `publish` or similar? `set_input` implies to me that an internal value is set, not resulting in any outgoing communication, making it easy to forget to handle the step or assume it's empty.

Reply:
*e.g.* `propose` is used further down this PR.

Reply:
I amended the commit:

```diff
--- a/src/binary_agreement/binary_agreement.rs
+++ b/src/binary_agreement/binary_agreement.rs
@@ -77,7 +77,7 @@ impl<N: NodeIdT> DistAlgorithm for BinaryAgreement<N> {
     type Error = Error;
 
     fn handle_input(&mut self, input: Self::Input) -> Result<Step<N>> {
-        self.set_input(input)
+        self.propose(input)
     }
 
     /// Receive input from a remote node.
@@ -120,14 +120,14 @@ impl<N: NodeIdT> BinaryAgreement<N> {
         })
     }
 
-    /// Sets the input value for Binary Agreement.
+    /// Proposes a boolean value for Binary Agreement.
     ///
-    /// If more than two thirds of validators input the same value, that will eventually be output.
-    /// Otherwise either output is possible.
+    /// If more than two thirds of validators propose the same value, that will eventually be
+    /// output. Otherwise either output is possible.
     ///
-    /// Note that if `can_input` returns `false`, it is already too late to affect the outcome.
-    pub fn set_input(&mut self, input: bool) -> Result<Step<N>> {
-        if !self.can_input() {
+    /// Note that if `can_propose` returns `false`, it is already too late to affect the outcome.
+    pub fn propose(&mut self, input: bool) -> Result<Step<N>> {
+        if !self.can_propose() {
             return Ok(Step::default());
         }
         // Set the initial estimated value to the input value.
@@ -157,7 +157,7 @@ impl<N: NodeIdT> BinaryAgreement<N> {
 
     /// Whether we can still input a value. It is not an error to input if this returns `false`,
     /// but it will have no effect on the outcome.
-    pub fn can_input(&self) -> bool {
+    pub fn can_propose(&self) -> bool {
         self.epoch == 0 && self.estimated.is_none()
     }
```